src/setEnvVars.ts

```diff
@@ -0,0 +1,83 @@
+import path from 'path';
+import { existsSync, readFileSync } from 'fs';
+import dotenv from 'dotenv';
+import dotenvExpand from 'dotenv-expand';
+import { getNextConfig } from './nextConfig';
+import { RuntimeEnvironment } from './constants';
+const { SERVER, CLIENT } = RuntimeEnvironment;
+
+// @NOTE Next.js env var handling implementation is available here:
+// https://github.com/vercel/next.js/tree/v10.0.5/test/integration/env-config
+// We can't use it as long as this doesn't get fixed:
+// https://github.com/vercel/next.js/issues/21296
+let dotFile: Record<string, string> | undefined = undefined;
+export function loadDotFile({ nextRoot }: { nextRoot: string }) {
+  // https://github.com/vercel/next.js/blob/v10.0.5/packages/next-env/index.ts#L82
+  const dotFilePaths = [
```

Comment:
Would it make sense to let user pass the `dotfile` path? I've noticed I'm loading a different file in my project (did it manually for now) `.env.development` https://github.com/insights-io/Insight/blob/master/frontend/app/src/__tests__/setup.ts#L7 and it would be annoying if I had to copy same contents of the file for `next-page-tester`  to load it.

Reply:
This would also be a decent solution, even simpler than what we have now. Maybe a relative path from `nextRoot`?

If we exposed a `dotfilePath` option I'd probably get rid of default `dotFilePaths`. Just to not increase complexity exponentially. How do you see it?

Reply:
Hmm, I think it still makes sense to fallback to those values. We can implement in a super simple way by just pushing the user `dotfilePath` in front.

```
if(dotfilePath) {
   dotFilePaths.unshift(dotfilePath)
|
```

Reply:
I was considering edge cases like  having a `.env.test.local` in place (maybe for e2e tests) and preferring to run next-page-tester with no dotfiles at all (maybe declaring env var in-line).

A custom `dotfilePath` would cover any scenario at the cost of a bit more repetition. Otherwise I guess we would and up by forcing `dotfilePath` with a `false` boolean to disable dotfiles lookup.

Reply:
If I understand correctly you are thinking about the option to disable `dotfile` loading.

We can make `dotfile?: string | false`:
- If nothing is passed; behaviour is as now
- If `false` is passed; dotfile loading is skipped
- If `string` is passed; path is added to 1st index of our `dotfilePath`

This seems to be flexible enough and covers all usecases?

About the repetition; Its up to user to handle it, but personally I would just wrap the `getPage` (and would encourage others to do so as well) to specify props used in all tests (e.g. `dotfile` , `useDocument`,...)


```js
export const getPage = (options) => {
   return getPageImpl({
       dotfile: '.env.development',
       useDocument: true,
       ...options,
     }
   )
}
```

Reply:
Sorry @Meemaw, I was quite cryptic in my last comment. :)

Actually I was considering the opportunity to avoid introducing convoluted API's (like `dotfile?: string | false` + default loading behaviour) by dropping autoloading (what we do now) and only load the dotfile provided (if provided) via options.

Would it an acceptable solution?

Reply:
Yeah, It works fine for me and it seems as a more "flexible" approach. We can always add the auto-loading back if we get a use-case/request/issue that makes sense.